File: src/rails/game/model/CertificateCountModel.java
package rails.game.model;

import rails.game.state.Model;

public final class CertificateCountModel extends Model {
    
    public final static String ID = "CertificateCountModel"; 

    private CertificateCountModel(PortfolioModel parent) {
        super(parent, ID);
    }
  
    public static CertificateCountModel create(PortfolioModel parent){
        CertificateCountModel model = new CertificateCountModel(parent);
        // lets certificate count model update on portfolio changes
        parent.addModel(model);
        return model;
    }

    /**
     * @return restricted to PortfolioModel
     */
    @Override
    public PortfolioModel getParent() {
        return (PortfolioModel)super.getParent();
    }

    @Override
    public String observerText() {
        return ("" + getParent().getCertificateCount()).replaceFirst("\\.0", "").replaceFirst("\\.5", "\u00bd");
    }

}

File: dotnet/RailsLib.Net/Net/Game/Model/CertificateCountModel.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.RegularExpressions;

namespace GameLib.Net.Game.Model
{
    public class CertificateCountModel : RailsModel
    {
        public const string ID = "CertificateCountModel";

        private CertificateCountModel(PortfolioModel parent) : base(parent, ID)
        {
        }

        public static CertificateCountModel Create(PortfolioModel parent)
        {
            CertificateCountModel model = new CertificateCountModel(parent);
            // lets certificate count model update on portfolio changes
            parent.AddModel(model);
            return model;
        }

        /**
         * @return restricted to PortfolioModel
         */
        new public PortfolioModel Parent
        {
            get
            {
                return (PortfolioModel)base.Parent;
            }
        }

        override public string ToText()
        {
            Regex r1 = new Regex("\\.0");
            Regex r2 = new Regex("\\.5");
            return r1.Replace(r2.Replace(Parent.GetCertificateCount().ToString(), "", 1), "\u00bd", 1);
        }
    }
}
